asynch/proto/connection.py

```diff
@@ -564,16 +562,16 @@ def reset_state(self):
 
     async def disconnect(self):
         if self.connected:
-            self.connected = False
             await self.writer.close()
-        self.reset_state()
+            self.reset_state()
+            self.connected = False
 
     async def connect(self):
         if self.connected:
             await self.disconnect()
-        logger.debug("Connecting. Database: %s. User: %s", self.database, self.user)
+        logger.debug(f"Connecting. Database: {self.database}. User: {self.user}")
```

Comment:
nit: Were these changes automatic or intentional? Generally it's preferred not to use explicit formatting in logging, as it will be executed on `logger.debug` instead of later when specific logging level was picked for propagation. 
Not important, same for the line below